Gems/NvCloth/Code/Source/Pipeline/SceneAPIExt/ClothRule.cpp

```diff
@@ -35,9 +35,9 @@ namespace NvCloth
 
         AZStd::vector<AZ::Color> ClothRule::ExtractClothData(
             const AZ::SceneAPI::Containers::SceneGraph& graph,
+            const AZ::SceneAPI::Containers::SceneGraph::NodeIndex meshNodeIndex,
             const size_t numVertices) const
```

Comment:
From FBX Settings the user selected the mesh node for cloth in the ClothRule and it's stored there, so the cloth rule must be able to find the node it needs, not being passed as argument what node it needs. With this information, passing the mesh node to this function does not make sense.

The root issue is Mesh Optimizer mangling with the mesh node names, instead of just replacing the mesh from the existing nodes. Because of that, now every logic that selects nodes get affected, including this cloth rule logic.

Please remove passing the mesh node index to `ExtractClothData` and instead add the logic "using the optimized version if exists". It's unfortunate that this logic has to be spread, but that was an early decision made and we have to live with it now.